hpx/util/cache_aligned_data.hpp

```diff
@@ -10,8 +10,19 @@
 #include <hpx/config.hpp>
 #include <hpx/runtime/threads/topology.hpp>
 
-namespace hpx { namespace util
-{
+namespace hpx { namespace util {
+    namespace detail {
+        // Computes the padding required to fill up a full cache line after
+        // data_size bytes.
+        constexpr std::size_t get_cache_line_padding_size(
+            std::size_t data_size)
+        {
+            return (threads::get_cache_line_size() -
```

Comment:
```
    template <typename Data>
    struct cache_aligned_data
    {
        static constexpr unsigned int padding1 = sizeof(Data) % threads::get_cache_line_size();
        static constexpr unsigned int padding2 = (padding1==0) ? 0 : threads::get_cache_line_size()- padding1;
        
        constexpr std::size_t padding() const {
            return padding2;
        }
        
        // pad to cache line size bytes
        Data data_;
        char cacheline_pad[padding2];        
    };
```

Reply:
I use an extra check to pad with zero if the cache is aligned already (modulus returns zero). then if you have a 64 byte struct or something, it doesn't use 128 bytes.

Reply:
This calculation has the same effect (that's the whole point of this PR). But as I said I'll change it to something more explicit like your version if it's not very clear from the way it's written now. Is that what you're suggesting?

Reply:
sorry. I misread the code. Looks fine.